plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p3order/ForsterConstraintResolver.java

```diff
@@ -58,10 +58,10 @@
      *            the current node order
      */
     public ForsterConstraintResolver(final LNode[][] currentNodeOrder) {
-        assert currentNodeOrder.length > 0 && currentNodeOrder[0].length > 0;
-        
-        constraintsBetweenNonDummies = currentNodeOrder[0][0].getGraph().getProperty(
-                InternalProperties.IN_LAYER_SUCCESSOR_CONSTRAINTS_BETWEEN_NON_DUMMIES);
+        if (currentNodeOrder.length > 0 && currentNodeOrder[0].length > 0) {
+            constraintsBetweenNonDummies = currentNodeOrder[0][0].getGraph()
+                    .getProperty(InternalProperties.IN_LAYER_SUCCESSOR_CONSTRAINTS_BETWEEN_NON_DUMMIES);
+        }
```

Comment:
Ouch... Yeah, I didn't think there were empty graphs anywhere.

Reply:
To be honest, I would prefer not to create the `GraphInfoHolder` at all for an empty graph. However, I didn't see an easy way to adjust the code that way.